core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -177,9 +178,13 @@ object Parse {
 
     val parts = coords.split(":", 5)
 
-    val attributes = attrs.get("classifier") match {
-      case Some(c) => Attributes("", c)
-      case None => Attributes("", "")
+    val attributes = Attributes("", attrs.getOrElse("classifier", ""), decode(attrs.getOrElse("url", ""), "UTF-8"))
```

Comment:
While you are at it, could you add a check to throw an error if `attrs` contains things other than `classifier` or `url`?